homeassistant/config_entries.py

```diff
@@ -2709,6 +2709,20 @@ def async_create_entry(  # type: ignore[override]
         options: Mapping[str, Any] | None = None,
     ) -> ConfigFlowResult:
         """Finish config flow and create a config entry."""
+        if self.source in {SOURCE_REAUTH, SOURCE_RECONFIGURE}:
+            report_issue = async_suggest_report_issue(
+                self.hass, integration_domain=self.handler
+            )
+            _LOGGER.warning(
+                (
+                    "Detected %s config flow using `async_create_entry`, which "
+                    "can cause unexpected behavior and will stop working in %s. "
```

Comment:
I think we should not say "can cause unexpected behavior", I think we should say we want reconfigure and reauth flows to update an existing entry and then abort, not create a new entry.

Reply:
Message updated